src/stcal/ramp_fitting/ramp_fit.py

```diff
@@ -212,9 +229,47 @@ def ramp_fit_data(ramp_data, buffsize, save_opt, readnoise_2d, gain_2d,
         nframes = ramp_data.nframes
         readnoise_2d *= gain_2d / np.sqrt(2. * nframes)
 
+        # Suppress one group ramps, if desired.
+        if ramp_data.suppress_one_group_ramps:
+            suppress_one_group_ramps(ramp_data)
+
         # Compute ramp fitting using ordinary least squares.
         image_info, integ_info, opt_info = ols_fit.ols_ramp_fit_multi(
             ramp_data, buffsize, save_opt, readnoise_2d, gain_2d, weighting, max_cores)
         gls_opt_info = None
 
     return image_info, integ_info, opt_info, gls_opt_info
+
+
+def suppress_one_group_ramps(ramp_data):
+    """
+    Finds one group ramps in each integration and suppresses them, i.e. turns
+    them into zero group ramps.
+
+    Parameter
+    ---------
+    model : data model
+        input data model, assumed to be of type RampModel
+    """
+    dq = ramp_data.groupdq
+    nints, ngroups, nrows, ncols = dq.shape
+
+    for k in range(nints):
+        intdq = dq[k, :, :, :]
+        good_groups = np.zeros(intdq.shape, dtype=int)
+
+        # Compute the number of good groups in each ramp
+        good_groups[intdq == 0] = 1
```

Comment:
I'm not sure that relying on `intdq` being exactly zero to indicate a good group is entirely appropriate. It's possible that for some flag values to be present (non-zero) that are to be treated as information only (e.g. a warning), but does not render the data unusable. It should be only cases where a group has the DO_NOT_USE flag set that they should be considered completely bad. This assumes that any group that has either the SATURATED or JUMP flags set also has DO_NOT_USE set. We would need to check that.

Reply:
This will require discussion because this is how segments are computed at